Lib/turtle.py

```diff
@@ -2812,13 +2812,13 @@ def clone(self):
         joe = mick.clone()
         """
         screen = self.screen
-        self._newLine(self._drawing)
 
         turtle = self.turtle
         self.screen = None
         self.turtle = None  # too make self deepcopy-able
 
         q = deepcopy(self)
+        self._newLine(self._drawing)
```

Comment:
The doctest fails with 20 failures: https://github.com/python/cpython/actions/runs/11687342848/job/32545252133?pr=126401.  One may have to click the label "Search log" label to see the log.  This is the first failure.

File "library/turtle.rst", line 1688, in default
Failed example:
    joe = mick.clone()
Exception raised:
    Traceback (most recent call last):
      File "/home/runner/work/cpython/cpython/Lib/doctest.py", line 1395, in __run
        exec(compile(example.source, filename, "single",
        ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                     compileflags, True), test.globs)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "<doctest default[1]>", line 1, in <module>
        joe = mick.clone()
      File "/home/runner/work/cpython/cpython/Lib/turtle.py", line 2821, in clone
        self._newLine(self._drawing)
        ~~~~~~~~~~~~~^^^^^^^^^^^^^^^
      File "/home/runner/work/cpython/cpython/Lib/turtle.py", line 3367, in _newLine
        self.screen._drawline(self.currentLineItem, top=True)
        ^^^^^^^^^^^^^^^^^^^^^
    AttributeError: 'NoneType' object has no attribute '_drawline'

This failure seems to be caused by moving _drawline() to a location where self.screen has been set to None.  So I suspect that this line should be after self.screen is restored.

The remaining failures might be side-effects of this failure, but I don't know how doctest works.
@AA-Turner You recently patched turtle doctests.  Do you have any better insight?

Reply:
Line was moved back to whee it was and doctest passes.